nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -177,7 +180,10 @@ bool Controller::isTrajectoryCollisionFree(
       trajectory_pub_->publish(trajectory);
       return false;
     }
-  }
+
+    // Check if we reach the goal
+    distance = nav2_util::geometry_utils::euclidean_distance(target_pose, next_pose.pose);
+  }while(distance > 1e-2 || trajectory.poses.size() < max_iter);
```

Comment:
```suggestion
  }while(distance > 1e-2 && trajectory.poses.size() < max_iter);
```

Reply:
I'm an idiot this morning - you were right with &&. We want to do it while both are true. If any is false, we want to exit :facepal

Reply:
Yeah, I was writing the truth table and thinking myself "ehh, it was right the first time with the && ...". We need holidays 🤣